Sources/TuistCache/ContentHashing/TargetContentHasher.swift

```diff
@@ -20,46 +30,84 @@ public final class TargetContentHasher: TargetContentHashing {
-            coreDataModelsContentHasher: CoreDataModelsContentHasher(contentHasher: contentHasher)
+            coreDataModelsContentHasher: CoreDataModelsContentHasher(contentHasher: contentHasher),
+            resourcesContentHasher: ResourcesContentHasher(contentHasher: contentHasher),
+            headersContentHasher: HeadersContentHasher(contentHasher: contentHasher),
+            deploymentTargetContentHasher: DeploymentTargetContentHasher(contentHasher: contentHasher),
+            infoPlistContentHasher: InfoPlistContentHasher(contentHasher: contentHasher),
+            settingsContentHasher: SettingsContentHasher(contentHasher: contentHasher),
+            dependenciesContentHasher: DependenciesContentHasher(contentHasher: contentHasher)
         )
     }
 
     public init(
         contentHasher: ContentHashing,
         sourceFilesContentHasher: SourceFilesContentHashing,
         targetActionsContentHasher: TargetActionsContentHashing,
-        coreDataModelsContentHasher: CoreDataModelsContentHashing
+        coreDataModelsContentHasher: CoreDataModelsContentHashing,
+        resourcesContentHasher: ResourcesContentHashing,
+        headersContentHasher: HeadersContentHashing,
+        deploymentTargetContentHasher: DeploymentTargetContentHashing,
+        infoPlistContentHasher: InfoPlistContentHashing,
+        settingsContentHasher: SettingsContentHashing,
+        dependenciesContentHasher: DependenciesContentHashing
     ) {
         self.contentHasher = contentHasher
         self.sourceFilesContentHasher = sourceFilesContentHasher
         self.coreDataModelsContentHasher = coreDataModelsContentHasher
         self.targetActionsContentHasher = targetActionsContentHasher
+        self.resourcesContentHasher = resourcesContentHasher
+        self.headersContentHasher = headersContentHasher
+        self.deploymentTargetContentHasher = deploymentTargetContentHasher
+        self.infoPlistContentHasher = infoPlistContentHasher
+        self.settingsContentHasher = settingsContentHasher
+        self.dependenciesContentHasher = dependenciesContentHasher
     }
 
     // MARK: - TargetContentHashing
 
     public func contentHash(for targetNode: TargetNode) throws -> String {
         let target = targetNode.target
         let sourcesHash = try sourceFilesContentHasher.hash(sources: target.sources)
-        let resourcesHash = try hash(resources: target.resources)
+        let resourcesHash = try resourcesContentHasher.hash(resources: target.resources)
         let coreDataModelHash = try coreDataModelsContentHasher.hash(coreDataModels: target.coreDataModels)
         let targetActionsHash = try targetActionsContentHasher.hash(targetActions: target.actions)
-        let stringsToHash = [sourcesHash,
-                             target.name,
+        let dependenciesHash = try dependenciesContentHasher.hash(dependencies: target.dependencies)
```

Comment:
You are hashing the dependency reference, not the dependency itself. For example, if target A says it depends on B, you are doing `hash("A depends on B")`, when it should be `hash(B)`. To get the actual dependency, you have to get it from `targetNode`: `targetNode.dependencies`.

Reply:
Yep, it's a conscious choice:
if we are hashing `A`, a change in `B` shouldn't invalidate `A` while a change in the dependency from `A` to `B` should. Does it make sense? Why wouldn't you use the cached version of `A` if `B` changes?

Reply:
I leave this for the next PR. It's still progress to consider more info into the hash. I want to test to see what happens with linking when a cached dependency breaks its public API